controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -95,20 +175,32 @@ void wfa_ca::handle_wfa_ca_message(
         return;
     }
 
-    const char *command_ptr = request->command();
+    std::string command_str(request->command());
 
-    LOG(DEBUG) << "Received ACTION_BML_WFA_CA_CONTROLLER_REQUEST, command=" << command_ptr;
+    LOG(DEBUG) << "Received ACTION_BML_WFA_CA_CONTROLLER_REQUEST, command=\"" << command_str
+               << "\"";
 
-    auto cmd_tokens_vec = string_utils::str_split(command_ptr, ',');
+    std::string err_string;
 
-    if (cmd_tokens_vec.size() < 1) {
-        LOG(ERROR) << "Invalid WFA-CA command";
-        reply_invalid_error(sd, cmdu_tx, eWfaCaStatus::INVALID, "empty command");
+    auto forbidden_chars = check_forbidden_chars(command_str);
+    if (forbidden_chars.empty()) {
+        err_string = "command contain illegal chars";
+        LOG(ERROR) << err_string << ": " << forbidden_chars;
+        reply_invalid_error(sd, cmdu_tx, eWfaCaStatus::INVALID, err_string);
+    }
+
+    g_cmd_tokens_vec = string_utils::str_split(command_str, ',');
+    g_token_idx      = 0;
```

Comment:
I think this is better as:

```
auto cmd_tokens_vec = string_utils::str_split(command_str, ',');

if (cmd_tokens_vec.empty()) {... error ...}

std::string command_type_str = cmd_tokens_vec.begin();
cmd_tokens_vec.erase(cmd_tokens_vec.begin());
...
parse_params(cmd_tokens_vec);
```

Reply:
I really believe this is a lot cleaner and easier to understand than the explicit indexing in a global variable, even if it requires calling std::next inside the loop inside the parse_params function. The current code basically re-implements a standard iterator.

Reply:
fine I changed it. But I really believe that what I did looks better :\